src/main/java/org/embl/mobie/io/ome/zarr/loaders/N5OMEZarrImageLoader.java

```diff
@@ -322,16 +322,20 @@ private void open() {
     private ArrayList<ViewRegistration> createViewRegistrations(int setupId, int setupTimePoints) {
         OmeZarrMultiscales multiscales = setupToMultiscale.get(setupId);
         AffineTransform3D transform = new AffineTransform3D();
-        if (multiscales.datasets[setupId].transformations != null && multiscales.datasets[setupId].transformations[0].scale != null) {
-            double[] scale = multiscales.datasets[setupId].transformations[0].scale;
-            if (scale.length > 2) {
-                transform.scale(scale[0] / 1000, scale[1] / 1000, scale[2] / 1000);
-            } else {
-                transform.scale(scale[0] / 1000, scale[1] / 1000, 1);
+        if (multiscales.datasets[setupId].coordinateTransformations != null) {
```

Comment:
Yes, but previous versions do not contain coordinateTransformation, thus for opening v2 and v3 we need these checks.

Reply:
I was gonna raise the same point.
I don't know what's best here.
One could consider checking the version right at the start and then jumping to completely different code, e.g. have it in the package structure: `org.embl.mobie.io.ome.zarr.v03`.
@joshmoore how are you guys handling this?

Reply:
> One could consider checking the version right at the start and then jumping to completely different code, e.g. have it in the package structure: `org.embl.mobie.io.ome.zarr.v03`.

I think the proper way to go is having a separate interface for the metadata, that can be used here and exposes `getAffine` methodes etc. to get the necessary information. Then this interface can be implemented for the different ome.zarr versions.

This is essentially what n5-imglib2.metadata does, so instead of reimplementing the logic here we would rather use it and contribute our parsers for ome.zarr to it. We have discussed this with @bogovicj during the hackathon already and I will follow up on this once we have 0.4 fully released.

Reply:
Yes, I think we can discuss this better once we actually see how such an Interface would look like.